### ISSUE TITLE
nx-cloud agents fail due to override arguments getting treated as project names

### ISSUE THREAD

**Issue by Phault** [nrwl/nx#14004]
### Current Behavior

Running the following with an Nx cloud-enabled workspace (note that `--ci` is before `--target` to account for #14002):

```shell
pnpm exec nx-cloud start-ci-run
NX_CLOUD_DISTRIBUTED_EXECUTION=true pnpm exec nx affected --ci --target=test
```

Will spit out the following on the executing agent:

```
nx.js run-many

Run target for multiple listed projects

<snipped for brevity>

Find more information and examples at https://nx.dev/nx/run-many

Error: No projects matching: --ci
```

It appears that the agent invokes run-many, but places the overrides somewhere after --project, which fails due to #14002. 

### Expected Behavior

The `--ci` argument should be treated as an override for the executed targets.

### Steps to Reproduce

1.

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node : 16.16.0
   OS   : darwin arm64
   pnpm : 7.19.0
   
   nx : 15.4.0
   @nrwl/angular : Not Found
   @nrwl/cypress : 15.4.0
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.4.0
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.4.0
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 15.4.0
   @nrwl/js : 15.4.0
   @nrwl/linter : 15.4.0
   @nrwl/nest : Not Found
   @nrwl/next : 15.4.0
   @nrwl/node : Not Found
   @nrwl/nx-cloud : 15.0.2
   @nrwl/nx-plugin : 15.4.0
   @nrwl/react : 15.4.0
   @nrwl/react-native : Not Found
   @nrwl/rollup : 15.4.0
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : 15.4.0
   @nrwl/webpack : 15.4.0
   @nrwl/workspace : 15.4.0
   @nrwl/vite : 15.4.0
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   	 nx-bitbucket-server
   	 nx-github
   	 nx-vercel
   ---------------------------------------
   Community plugins:
   	 @jscutlery/semver: 2.29.3
   	 ngx-deploy-npm: 4.3.10
```

**Comment by daniel-zero** [comment]
I can confirm this error in our project. We use the following command to run unit-tests
```
npx nx run-many --all --target=test --parallel=5 --ci --code-coverage --coverageReporters=lcov
``

**Comment by yharaskrik** [comment]
Also seeing this with the `test` target (using batch mode as well), moving all flags before the `--projects` flag fixes it.

**Comment by sebastiandg7** [comment]
CI run example with the problem: https://github.com/nxkit/nxkit/actions/runs/3784050589/jobs/6433033478#step:16:45

**Comment by StalkAltan** [comment]
Hi there, we are aware of this issue and are taking a look. Going to close this as it is effectively a duplicate of https://github.com/nrwl/nx/issues/14002. Please check there for further updates.

For the time being, downgrading to the latest 15.3 release should resolve the problem.

**Comment by StalkAltan** [comment]
Fix has been released as a part of 15.4.2
